src/source/source_tree.ml

```diff
@@ -194,6 +194,14 @@ and contents
   =
   let files = Dir_contents.files readdir in
   let+ dune_file = Dune_file.load ~dir:path dir_status project ~files ~parent:dune_file in
+  let files =
+    let predicate =
+      match dune_file with
+      | None -> Dune_file.Files.default
+      | Some dune_file -> Dune_file.files dune_file
+    in
+    Dune_file.Files.eval predicate ~files
```

Comment:
I think this means that we're filtering the file set even if no predicate was specified? Would be better not to do that.

Reply:
Good point. Fixed in cb30893b0ce738caac8c589da3f15775a493a27c.